### PR TITLE
refactor: use primordials for extensions/websocket

### PR DESCRIPTION
**Pull request by littledivy**
Ref #11224

### CODE REVIEW

File: extensions/websocket/01_websocket.js [hunk 1]
```diff
@@ -218,54 +250,67 @@
...
-
-            const event = new CloseEvent("close");
-            this.dispatchEvent(event);
-            tryClose(this[_rid]);
-          });
-        } else {
-          this[_readyState] = OPEN;
-          const event = new Event("open");
-          this.dispatchEvent(event);
+      PromisePrototypeCatch(
```
Comment (lucacasonato) [inline extensions/websocket/01_websocket.js:265]: You can combine the "catch" into the `PromisePrototypeThen` by passing the catch handler as a second callback.

**Review by lucacasonato — COMMENTED** [review]
Can you import `Blob` from `window.__bootstrap.file` for line 48?

File: extensions/websocket/01_websocket.js (hunk 1 above)
Comment (littledivy) [reply extensions/websocket/01_websocket.js]: done

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you Divy

**Pull request merged**